postgre-client/src/tasks.rs

```diff
@@ -16,47 +17,50 @@ pub const MAX_BUFFERED_TASKS_TO_TAKE: usize = 5000;
 
 pub struct UpdatedTask {
     pub status: TaskStatus,
+    pub mutability: String,
     pub metadata_url: String,
-    pub error: String,
+    pub etag: Option<String>,
+    pub last_modified_at: Option<DateTime<Utc>>,
 }
 
 #[derive(Debug, Clone)]
 pub struct JsonTask {
-    pub tsk_id: Vec<u8>,
+    pub metadata_hash: Vec<u8>,
     pub metadata_url: String,
     pub status: TaskStatus,
 }
 
 impl PgClient {
-    async fn insert_new_tasks(&self, data: &mut Vec<Task>) -> Result<(), IndexDbError> {
-        data.sort_by(|a, b| a.ofd_metadata_url.cmp(&b.ofd_metadata_url));
+    pub async fn insert_new_tasks(&self, data: &mut Vec<Task>) -> Result<(), IndexDbError> {
+        data.sort_by(|a, b| a.metadata_url.cmp(&b.metadata_url));
 
         let mut query_builder = QueryBuilder::new(
             "INSERT INTO tasks (
-                tsk_id,
-                tsk_metadata_url,
-                tsk_locked_until,
-                tsk_attempts,
-                tsk_max_attempts,
-                tsk_error,
-                tsk_status
+                metadata_hash,
+                metadata_url,
+                etag,
+                last_modified_at,
+                mutability,
+                next_try_at,
+                task_status
             ) ",
         );
 
-        query_builder.push_values(data, |mut b, off_d| {
-            let tsk = UrlWithStatus::new(off_d.ofd_metadata_url.as_str(), false);
-            b.push_bind(tsk.get_metadata_id());
-            b.push_bind(tsk.metadata_url);
-            b.push_bind(off_d.ofd_locked_until);
-            b.push_bind(off_d.ofd_attempts);
-            b.push_bind(off_d.ofd_max_attempts);
-            b.push_bind(off_d.ofd_error.clone());
-            b.push_bind(off_d.ofd_status);
+        query_builder.push_values(data, |mut b, offchain_data| {
+            let task = UrlWithStatus::new(offchain_data.metadata_url.as_str(), false);
+            b.push_bind(task.get_metadata_id());
+            b.push_bind(task.metadata_url);
+            b.push_bind(offchain_data.etag.clone());
+            b.push_bind(offchain_data.last_modified_at);
+            b.push_bind(offchain_data.mutability);
+            b.push_bind(offchain_data.next_try_at);
```

Comment:
Apart form the tests I didn't find where is the next_try_at being set, can you reference me to it?

Reply:
wierd naming I guess.
```#[derive(Debug, Clone, Default)]
pub struct Task {
    pub metadata_url: String,
    pub etag: Option<String>,
    pub last_modified_at: Option<DateTime<Utc>>,
    pub mutability: OffchainDataMutability,
    pub next_try_at: DateTime<Utc>,
    pub status: TaskStatus,
}```
offchain_data = Task in that context, should be renamed